apps/server/src/modules/files-storage/repo/filerecord.repo.ts

```diff
@@ -46,11 +51,50 @@ export class FileRecordRepo extends BaseRepo<FileRecord> {
 		return result;
 	}
 
-	async findBySecurityCheckRequestToken(token: string): Promise<FileRecord> {
+	public async findBySecurityCheckRequestToken(token: string): Promise<FileRecord> {
 		// Must also find expires in future. Please do not add .byExpires().
 		const scope = new FileRecordScope().bySecurityCheckRequestToken(token);
+		const fileRecord = await this.findOne(scope);
 
-		const fileRecord = await this.findOneOrFail(scope);
+		return fileRecord;
+	}
+
+	public async delete(fileRecords: FileRecord[]): Promise<void> {
+		const entities = await this.find(fileRecords);
+		await this.dbm.remove(entities);
+	}
+
+	// note: persist is split in update and create
+	public async update(fileRecords: FileRecord[]): Promise<FileRecord[]> {
+		const entities = await this.find(fileRecords);
```

Comment:
Same question as above: Do we perform updates on deleted filerecords? Should a query be added?